optimizely/config_manager.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2019, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import requests
+import threading
+import time
+from requests import codes as http_status_codes
+from requests import exceptions as requests_exceptions
+
+from . import exceptions as optimizely_exceptions
+from . import logger as optimizely_logger
+from . import project_config
+from .error_handler import NoOpErrorHandler as noop_error_handler
+from .helpers import enums
+
+
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
+
+
+class BaseConfigManager(ABC):
+    """ Base class for Optimizely's config manager. """
+
+    def __init__(self,
+                 logger=None,
+                 error_handler=None,
+                 **kwargs):
```

Comment:
Should we accept arbitrary `kwargs` on the base class?

Reply:
Not really. Removed.